core/src/main/java/org/infinispan/remoting/transport/jgroups/CustomRequestCorrelator.java

```diff
@@ -1,35 +1,50 @@
 package org.infinispan.remoting.transport.jgroups;
 
+import java.io.ByteArrayInputStream;
+
+import org.infinispan.commons.marshall.StreamingMarshaller;
 import java.util.concurrent.Executor;
 
 import org.jgroups.Address;
 import org.jgroups.View;
+import org.jgroups.blocks.Request;
 import org.jgroups.blocks.RequestCorrelator;
 import org.jgroups.blocks.RequestHandler;
 import org.jgroups.conf.ClassConfigurator;
 import org.jgroups.stack.Protocol;
+import org.jgroups.util.Util;
 
 /**
- * Extend {@link RequestCorrelator} to ignore {@link org.jgroups.Event#SUSPECT} events.
- *
- * {@link SuspectException} will only be thrown when the target actually leaves the view.
+ * Extend {@link RequestCorrelator} to use our own marshaller.
  *
  * @author Dan Berindei
- * @since 7.2
+ * @since 9.0
  */
-public class CustomRequestCorrelator extends RequestCorrelator {
+class CustomRequestCorrelator extends RequestCorrelator {
    private final Executor remoteExecutor;
+   private final StreamingMarshaller ispnMarshaller;
 
    public CustomRequestCorrelator(Protocol transport, RequestHandler handler, Address local_addr,
-                                  Executor remoteExecutor) {
+                                  Executor remoteExecutor, StreamingMarshaller ispnMarshaller) {
       // Make sure we use the same protocol id as the parent class
       super(ClassConfigurator.getProtocolId(RequestCorrelator.class), transport, handler, local_addr);
       this.remoteExecutor = remoteExecutor;
+      this.ispnMarshaller = ispnMarshaller;
    }
 
    @Override
-   public void receiveSuspect(Address mbr) {
-      // Ignore suspect events, if the node actually left that will be handled by receiveView(View)
+   protected void handleResponse(Request req, Address sender, byte[] buf, int offset, int length,
+                                 boolean is_exception) {
+      Object retval;
+      try {
+         ByteArrayInputStream is = new ByteArrayInputStream(buf, offset, length);
+         retval = ispnMarshaller.objectFromInputStream(is);
```

Comment:
Done, I should have realized that my unmarshalling was just as bad as my marshalling :)